src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -0,0 +1,488 @@
+
+            //Force this to return true for broker test app
+
+            return VerifyAuthenticator(_androidAccountManager)
+                   && !packageName.Equals(BrokerConstants.PackageName, StringComparison.OrdinalIgnoreCase)
+                   && !packageName
+           .Equals(BrokerConstants.AzureAuthenticatorAppPackageName, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public Intent GetIntentForInteractiveBrokerRequest(IDictionary<string, string> brokerPayload, Activity callerActivity)
+        {
+            Intent intent = null;
+
+            try
+            {
+                IAccountManagerFuture result = null;
+                // Callback is not passed since it is making a blocking call to get
+                // intent. Activity needs to be launched from calling app
+                // to get the calling app's metadata if needed at BrokerActivity.
+
+                Bundle addAccountOptions = new Bundle();
+                addAccountOptions.PutString(BrokerConstants.BrokerAccountManagerOperationKey, BrokerConstants.GetIntentForInteractiveRequest);
+
+                _logger.Info("Broker Account Name: " + GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Username));
+
+                result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
+                    BrokerConstants.AuthtokenType, null, addAccountOptions, null,
+                    null, GetPreferredLooper(callerActivity));
+
+                if (result == null)
+                {
+                    _logger.Info("Android account manager AddAccount didn't return any results. ");
+                }
+
+                Bundle bundleResult = (Bundle)result?.Result;
+                // Broker should throw OperationCanceledException if token is not available
+                intent = (Intent)bundleResult?.GetParcelable(AccountManager.KeyIntent);
+
+                //Validate that the intent was created succsesfully.
+                if (intent != null)
+                {
+                    _logger.Info("Intent created from BundleResult is not null. ");
+                    // Need caller info UID for broker communication
+                    intent.PutExtra(BrokerConstants.CallerInfoUID, Binder.CallingUid);
+                }
+                else
+                {
+                    _logger.Info("Intent created from BundleResult is null. ");
+                    throw new MsalException(MsalError.NullIntentReturnedFromBroker, MsalErrorMessage.NullIntentReturnedFromBroker);
+                }
+
+                intent = GetInteractiveBrokerIntent(brokerPayload, intent);
+            }
+            catch
+            {
+                _logger.Error("Error when trying to acquire intent for broker authentication.");
+                throw;
+            }
+
+            return intent;
+        }
+
+        public string GetAuthTokenSilently(IDictionary<string, string> brokerPayload, Activity callerActivity)
```

Comment:
nit: maybe a better name? this is w/broker right?

Reply:
Yes. ill add broker